ext/standard/basic_functions.c

```diff
@@ -518,6 +518,7 @@ PHP_RSHUTDOWN_FUNCTION(basic) /* {{{ */
 	 * to the value in startup environment */
 	if (BG(locale_changed)) {
 		setlocale(LC_ALL, "C");
+		zend_reset_lc_ctype_locale();
```

Comment:
Should you remove the `setlocale()` call directly above this?

Reply:
The setlocale call above that sets LC_ALL to "C", while this one only sets LC_CTYPE to "C.UTF-8". We do need to reset other categories as well.

Reply:
Thanks for the clarification. I missed the difference between `LC_ALL` and `LC_CTYPE`.